types/audit.d.ts

```diff
@@ -80,12 +81,26 @@ declare global {
       extendedInfo?: {[p: string]: any};
       /** Overrides scoreDisplayMode with notApplicable if set to true */
       notApplicable?: boolean;
-      /** A numeric value that has a meaning specific to the audit, e.g. the number of nodes in the DOM or the timestamp of a specific load event. More information can be found in the audit details, if present. */
-      numericValue?: number;
       /** Extra information about the page provided by some types of audits, in one of several possible forms that can be rendered in the HTML report. */
       details?: Audit.Details;
     }
 
+    /** The Audit.Product type for audits that do not return a `numericValue`. */
+    interface NonNumericProduct extends ProductBase {
+      numericValue?: never;
+    }
+
+    /** The Audit.Product type for audits that do return a `numericValue`. */
+    interface NumericProduct extends ProductBase {
+      /** A numeric value that has a meaning specific to the audit, e.g. the number of nodes in the DOM or the timestamp of a specific load event. More information can be found in the audit details, if present. */
+      numericValue: number;
+      /** The unit of `numericValue`, used when the consumer wishes to convert numericValue to a display string. */
```

Comment:
lets say this is a superset (pending to be a subset) of https://tc39.es/proposal-unified-intl-numberformat/section6/locales-currencies-tz_proposed_out.html#sec-issanctionedsimpleunitidentifier